src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5; //FIGURE_NUM - number of figure subclasses
+    private static final int MAX_RANDOM = 9;
+
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        int randomFigure = random.nextInt(NUMBER_OF_FIGURES);
+
+        switch (randomFigure) {
+            case 0:
+                figure = new Square(getRandomSide(), colorSupplier.getRandomColor());
+                break;
+            case 1:
+                figure = new Rectangle(getRandomSide(), getRandomSide(),
+                        colorSupplier.getRandomColor());
+                break;
+            case 2:
+                figure = new RightTriangle(getRandomSide(), getRandomSide(),
+                        colorSupplier.getRandomColor());
+                break;
+            case 3:
+                figure = new IsoscelesTrapezoid(getRandomSide(), getRandomSide(),
+                        getRandomSide(), colorSupplier.getRandomColor());
+                break;
+            default:
+                figure = new Circle(getRandomSide(), colorSupplier.getRandomColor());
+                break;
+        }
+
+        return figure;
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, Color.WHITE.name());

```

Comment:
It is necessary to transfer the value 10 to the class constant, because it may not be clear what 10 is, why 10. If there is a name like DEFAULT_VALUE, it will be a little clearer.